corehq/apps/hqwebapp/tasks.py

```diff
@@ -99,6 +105,26 @@ def mail_admins_async(self, subject, message, fail_silently=False, connection=No
         self.retry(exc=e)
 
 
+@periodic_task(run_every=crontab(minute=0, hour=0), queue='background_queue')
+def process_bounced_emails():
+    if settings.RETURN_PATH_EMAIL and settings.RETURN_PATH_EMAIL_PASSWORD:
+        try:
+            bounced_manager = BouncedEmailManager(
+                delete_processed_messages=True
+            )
+            bounced_manager.process_bounces()
+            bounced_manager.process_complaints()
+            bounced_manager.logout()
```

Comment:
do we always want to run `bounced_manager.logout()` if `bounced_manager` was successfully instantiated above? If so it might make sense to put it in a `finally` block, or achieve the same behavior by making `BouncedEmailManager` a context manager and using `with BouncedEmailManager(...) as bounced_manager:` with the equivalent of `bounced_manager.logout()` in the context manager's `__exit__`.

Reply:
will add in followup PR. thank you!

Reply:
note to self: done